### PR TITLE
feat(joyride): show guided tour upon cryostat upgrades

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Fixes: #970 

## Description of the change:

Add a hint banner for app layout. This checks a local storage key to see if it should show or not. Only showed when logged in.

## Motivation for the change:

See #970

### CODE REVIEW

**Comment by tthvo** [comment]
what do u think? Let me know if u want a different message or close banner (won't show again) if the users follow the link :))

**Comment by maxcao13** [comment]
Looks great! Should the banner be closed when users click the link as well?

Also I think the link should be `quick starts` instead of `quickstarts`?

**Comment by tthvo** [comment]
@andrewazores How about now? I added a local storage key to keep track of Cryostat front-end asset version. If the build version is higher than one saved in local storage, it will show the tour (includes step 8/9 talking about quickstarts).

Was trying to use `semver` package but ran into this issue: https://github.com/npm/node-semver/issues/554. So, I just wrote a simple replacement that expects a complete semver (`major-minor-patch`) (this also helps reducing bundle size): https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

File: src/app/build.json [hunk 1]
```diff
@@ -11,5 +11,6 @@
   "fileIssueUrl": "https://github.com/cryostatio/cryostat/issues/new?labels=user+report,bug&body=Affects+__REPLACE_VERSION__",
   "mailingListName": "Google Groups",
   "mailingListUrl": "https://groups.google.com/g/cryostat-development",
-  "licenseUrl": "https://github.com/cryostatio/cryostat/blob/main/LICENSE"
+  "licenseUrl": "https://github.com/cryostatio/cryostat/blob/main/LICENSE",
+  "version": "2.4.0-dev"
```
Comment (andrewazores) [inline src/app/build.json:15]: We'll need to add a step to our release documentation that this key needs to be updated.

**Comment by andrewazores** [comment]
Looks great.

image

Might be nice to put that new version string on this modal? "Welcome to Cryostat 2.3.0!"

**Comment by andrewazores** [comment]
^ That version string should probably also be present on the AboutModal.

**Comment by tthvo** [comment]
|About Modal | Guided tour | 
|---|---|
|Screenshot from 2023-04-20 23-43-25|Screenshot from 2023-04-20 23-43-09|

**Comment by tthvo** [comment]
The version field in the about modal is off due to #882. Are we planning a fix for that for this release?

File: src/app/build.json (hunk 1 above)
Comment (tthvo) [reply src/app/build.json:15]: Yep! Updated our Upstream Release Process Docs now :))

**Comment by andrewazores** [comment]
> The version field in the about modal is off due to #882. Are we planning a fix for that for this release?

I'll take a look at it again. I think Max was probably right and simply tagging the main branch at the same place as the release branchpoint would probably fix it. If so, that's pretty simple and easy.

**Comment by andrewazores** [comment]
> > The version field in the about modal is off due to #882. Are we planning a fix for that for this release?
> 
> I'll take a look at it again. I think Max was probably right and simply tagging the main branch at the same place as the release branchpoint would probably fix it. If so, that's pretty simple and easy.

On current `main`:

image

On current `cryostat-v2.3`:

image

(I think it's `-dirty` because I forgot to `git submodule update` before this, so there are "unstaged changes" because of that)

Looks right. All I did was tag [the last common commit](https://github.com/cryostatio/cryostat/commit/ab46529dc7f6ebf9b8bf9105e5c2730f0d6f121f) before [your branch separation commit](https://github.com/cryostatio/cryostat/commit/485ed685ca38bd69c8cf0c69111ad0529af2cb0a) as the `v2.3.0` branchpoint.

**Review by andrewazores — APPROVED** [review]

**Pull request merged**